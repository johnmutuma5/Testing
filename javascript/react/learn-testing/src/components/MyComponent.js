import React from 'react';

const myComponent = (props) => {
    return (
        <div className='myComp'>
            <p className='para'> This is a paragraph </p>
            <SubComponent>Subcomponents contents</SubComponent>
        </div>
    );
};


const SubComponent = (props) => {
    return (
        <div className='sub'>
            { props.children }
            <SubSubComponent>Subsub contents</SubSubComponent>
        </div>
    );
};

const SubSubComponent = (props) => {
    return (
        <div className='subsub'>
            { props.children }
        </div>
    );
};

export default myComponent;
