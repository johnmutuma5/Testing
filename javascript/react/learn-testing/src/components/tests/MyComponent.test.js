import React from 'react';
import renderer from 'react-test-renderer';
import MyComponent from '../MyComponent';


describe("MyComponent TestCase", () => {
    test("MyComponent renders correctly", () => {
        const tree = renderer
            .create(<MyComponent />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
