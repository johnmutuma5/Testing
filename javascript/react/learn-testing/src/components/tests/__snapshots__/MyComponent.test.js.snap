// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent TestCase MyComponent renders correctly 1`] = `
<div
  className="myComp"
>
  <p
    className="para"
  >
     This is a paragraph 
  </p>
  <div
    className="sub"
  >
    Subcomponents contents
    <div
      className="subsub"
    >
      Subsub contents
    </div>
  </div>
</div>
`;
